scipy/ndimage/tests/test_filters.py

```diff
@@ -2982,3 +2986,44 @@ def test_edge_cases(self, xp):
         res = ndimage.vectorized_filter(input, function, size=21)
         ref = ndimage.vectorized_filter(input, function, size=21)
         xp_assert_close(res, ref)
+
+
+@pytest.mark.slow
+@given(
+    arr=npst.arrays(
+        dtype=np.float64,
+        shape=st.integers(min_value=1, max_value=20).map(lambda x: x * 50),
+    ),
+    size=st.integers(min_value=1, max_value=25),
+    mode=st.sampled_from(['reflect', 'constant', 'nearest', 'mirror', 'wrap']),
+)
+def test_gh_22586_median_filter_no_segfault_and_1d_matches_nd(
```

Comment:
I'm not seeing a fail before/pass after application of the patch with this test in its current state on x86_64 Linux, even if I manually increase the 
 number of scenarios probed via `@settings(max_examples=5000)`.

Is anyone seeing this test fail on Linux if the source patch is reverted back to `main`?

In my hands, the original reproducing test from gh-22586 still does the trick. I suppose we could add both tests to be safe perhaps, but I don't even see a failure let alone a segfault for this test on `main` at the moment.

Reply:
I see the same thing and after some digging, it appears I may have made a mistake when converting the test to hypothesis? If you drop that commit, the parametrized pytest version segfaults if the changes to `_rank_filter_1d.cpp` aren't applied.